products/dialogflow/terraform.yaml

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Overrides::ResourceOverrides
+  Agent: !ruby/object:Overrides::Terraform::ResourceOverride
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "dialogflow_agent_full"
+        primary_resource_id: "full_agent"
+        vars:
+          agent_name: "tf-test-full-agent"
+    id_format: "{{project}}"
+    import_format: ["{{project}}"]
+    # Skip sweeper gen due to non-standard urls.
+    skip_sweeper: true
+    properties:
+      description: !ruby/object:Overrides::Terraform::PropertyOverride
+        validation: !ruby/object:Provider::Terraform::Validation
+          function: 'validation.StringLenBetween(0, 500)'
+      avatarUri: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+      avatarUriBackend: !ruby/object:Overrides::Terraform::PropertyOverride
+        api_name: avatarUri
```

Comment:
I'm not sure what's going on with the URI here. Can you give me a quick explanation?

Reply:
If the user adds a valid image url on create/update, the backend will store the image in a bucket and return the bucket location as the new `avatarUri`. I set a computed field `avatarUriBackend` for this bucket location, and set the original `avatarUri` as ignore_read so it doesn't diff unless the user submits a new url